test/check-machines-networks

```diff
@@ -17,24 +17,17 @@
 # You should have received a copy of the GNU Lesser General Public License
 # along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
 
+import json
 import xml.etree.ElementTree as ET
 
 import machineslib
 import testlib
 from machinesxmls import TEST_NETWORK2_XML, TEST_NETWORK3_XML, TEST_NETWORK4_XML, TEST_NETWORK_IPV6_XML, TEST_NETWORK_XML
 
 
-def getNetworkDevice(m):
-    net_devices_str = m.execute("virsh iface-list")
-    net_devices_str = net_devices_str.split("\n", 2)[2]  # Remove first 2 lines of table header
-    if net_devices_str not in ["\n", "\r\n"]:
-        device = net_devices_str.split(' ', 2)[1]  # Get the name of device, ignoring spacing before device string
-    else:  # If $virsh-iface list did not return any device, check virsh nodedev-list net:noh
-        net_devices_str = m.execute("virsh nodedev-list net")
-        net_devices_str = net_devices_str.split("\n", 2)[0]
-        device = net_devices_str.split("_", 2)[1]  # Ignore prefix (example: net_enp0s31f6_8c_16_45_5f_77_34)
-
-    return device
+def getNetworkDefaultRouteDevice(m):
+    routes = json.loads(m.execute("ip --json route show default"))
```

Comment:
Is `show default` needed here? At least on fedora 40 and tumbleweed, the test will pass if you use `ip --json route`

Reply:
Yes, but only by happenstance -- as soon as you have more than one route, and the default one isn't the first one, that will fail. I'd rather make it explicit.

Reply:
I see. That's a fair point

Reply:
It seems the problem is that there is no default route on those test systems at this point. The previous code did not pick a specific interface, just any first one, not even enabled. On one system here the first virsh command would have picked "lo". Does routing to outside the host actually need to work? If yes, then which destination? As example 127.0.0.1 then `ip --json route get 127.0.0.1`. If not then maybe hard coding "lo" is fine.

Reply:
Ah, indeed -- [it failed](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1782-f22e9226-20240827-042551-fedora-40/log.html#88), as I locally tested in an inadequate way.

My main issue is that I have not the slightest idea what this operation in the test actually does --  it creates a libvirt network with a `<dev>` node, which happens to be "eth0" or "eth1". I *assume* this creates a bridge which forwards its traffic to the given host device? But then "lo" doesn't make much sense, does it?

So indeed I'll just drop the `default` here, that'll also pick "eth0". I don't know what exactly this does, but at least it stays compatible :sweat_smile: